### PR TITLE
Feat: `jenkins` Plugin Refactoring and Document Enhancement

### PR DESCRIPTION
**Pull request by daniel-hutao**
## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests

## Description

1. Remove password getting logic;
2. Remove jenkinsOptions;
3. Refine the document(Chinese version).

@IronCore864 You do not need to rush to improve the English documentation, because the current documentation is not yet stable, I may continue to add some content in the near future.

## Related Issues

#993

## New Behavior (screenshots if needed)

Main changes: Document enhancement

### CODE REVIEW

File: docs/plugins/harbor.zh.md [hunk 1]
```diff
@@ -159,7 +159,7 @@ kubectl port-forward -n harbor service/harbor --address=${ip} 80
 
 `harbor` 插件的配置项多数都有默认值，具体默认值信息如下表：
 
-| key                | default value            | description                         |
+| 键                 | 默认值                    | 描述                                 |
```
[the remark this answers is not in the document]
Comment (aFlyBird0) [reply docs/plugins/harbor.zh.md:162]: 或者“配置项”？

File: docs/plugins/jenkins.zh.md [hunk 2]
```diff
@@ -1,35 +1,265 @@
...
+- 有一个可用的 Kubernetes 集群，版本 1.10+
+- 配置好 StorageClass
+
+**可选满足**
+
+- 配置好 Ingress 控制器（如果需要使用 Ingress 暴露服务）
+
+如果你还没有准备好一个满足上述要求的 Kubernetes 集群，可以参考 [minikube 文档](https://minikube.sigs.k8s.io/docs/start/) 快速创建一个 Kubernetes 测试集群。
+在成功执行完 `minikube start` 命令后，假如需要启用 Ingress，可以通过 `minikube addons enable ingress` 命令完成 Ingress 控制器的启用。
+因为 minikube 方式部署的 Kubernetes 集群会自带一个名字为 standard 的 default StorageClass，所以当前集群满足上述全部前置要求。
```
Comment (aFlyBird0) [inline docs/plugins/jenkins.zh.md:21]: kind is the same.

File: docs/plugins/jenkins.zh.md [hunk 3]
```diff
@@ -1,35 +1,265 @@
...
+  options:
+    chart:
+      # custom configuration. You can refer to [Jenkins values.yaml](https://github.com/jenkinsci/helm-charts/blob/main/charts/jenkins/values.yaml)
+      values_yaml: |
+        serviceAccount:
+          create: true
+          name: jenkins
+        controller:
+          adminUser: "admin"
+          adminPassword: "changeme"
```
Comment (aFlyBird0) [inline docs/plugins/jenkins.zh.md:49]: cool!

**Review by aFlyBird0 — APPROVED** [review]

File: docs/plugins/jenkins.zh.md (hunk 2 above)
Comment (daniel-hutao) [reply docs/plugins/jenkins.zh.md:21]: 嗯；这里就只介绍 minikube 吧；kind 更多用在 e2e 环境里，本地测试还是 minikube 流行 and 好用

**Comment by aFlyBird0** [comment]
image
image

**Pull request merged**